tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/TeamMailboxMemberSetMethodContract.scala

```diff
@@ -0,0 +1,716 @@
+           |	"using": ["urn:ietf:params:jmap:core", "urn:ietf:params:jmap:mail", "com:linagora:params:jmap:team:mailboxes"],
+           |	"methodCalls": [
+           |		[
+           |			"TeamMailboxMember/set",
+           |			{
+           |				"accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+           |				"update": {
+           |            "${teamMailbox.asString()}": {
+           |                "${CEDRIC.asString()}": {"role": "invalid"}
+           |            }
+           |        }
+           |			},
+           |			"c1"
+           |		]
+           |	]
+           |}""".stripMargin)
+      .when
+      .post
+      .`then`
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .body("methodResponses[0]", jsonEquals(
+        s"""[
+           |    "TeamMailboxMember/set",
+           |    {
+           |        "accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+           |        "updated": {},
+           |        "notUpdated": {
+           |            "${teamMailbox.asString()}": {
+           |                "type": "invalidPatch",
+           |                "description": "Role could only be manager or member"
+           |            }
+           |        }
+           |    },
+           |    "c1"
+           |]""".stripMargin).withOptions(ImmutableList.of(IGNORING_ARRAY_ORDER)))
+  }
+
+  @Test
+  def updateShouldReturnNotUpdatedWhenUserIsNotMemberOfTheTeamMailbox(server: GuiceJamesServer): Unit = {
+    val teamMailbox = TeamMailbox(DOMAIN, TeamMailboxName("hiring"))
+
+    server.getProbe(classOf[TeamMailboxProbe])
+      .create(teamMailbox)
+      .addManager(teamMailbox, ANDRE)
+
+    `given`
+      .body(
+        s"""{
+           |	"using": ["urn:ietf:params:jmap:core", "urn:ietf:params:jmap:mail", "com:linagora:params:jmap:team:mailboxes"],
+           |	"methodCalls": [
+           |		[
+           |			"TeamMailboxMember/set",
+           |			{
+           |				"accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+           |				"update": {
+           |            "${teamMailbox.asString()}": {
+           |                "${CEDRIC.asString()}": {"role": "member"}
+           |            }
+           |        }
+           |			},
+           |			"c1"
+           |		]
+           |	]
+           |}""".stripMargin)
+      .when
+      .post
+      .`then`
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .body("methodResponses[0]", jsonEquals(
+        s"""[
+           |    "TeamMailboxMember/set",
+           |    {
+           |        "accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+           |        "updated": {},
+           |        "notUpdated": {
+           |            "${teamMailbox.asString()}": {
+           |                "type": "invalidPatch",
+           |                "description": "Wrong teamMailboxName ${teamMailbox.asString()}"
```

Comment:
We should return "not found" in order not to disclose the existance on the team mailbox.

I did not find the test were we try to add a member for a valid team mailbox name that do not exist.

Also the message is confusing here: after all the name is valid...